src/agent/onefuzz-agent/src/local/common.rs

```diff
@@ -162,6 +167,30 @@ pub fn get_synced_dirs(
     Ok(dirs?)
 }
 
+fn recursive_remove(path: &Path) -> Result<()> {
```

Comment:
It seems like this is recreating `remove_dir_all()`, while also calling it. What's the motivation here?

AFAICT, the differences are that we (1) include paths in some of the error messages, when recursively deleting, (2) we allow each recursive remove call to possibly fail one time due to some sort of race.

Reply:
The issue I was experiencing was due to Windows recursion.  I've updated the impl to just use the [remove_dir_all](https://docs.rs/remove_dir_all/0.7.0/remove_dir_all/) crate which handles the issues in a more coherent fashion.